pal/system-info/minigame/system-info.ts

```diff
@@ -117,24 +119,38 @@ class SystemInfo extends EventTarget {
             [Feature.EVENT_HMD]: this.isXR,
         };
 
+        this._initPromise = [];
+        this._initPromise.push(this._supportsWebpPromise());
+
         this._registerEvent();
     }
 
-    private _supportsWebp (): boolean {
-        // NOTE: canvas.toDataURL() is not supported on WeChat iOS end (Found on iPhone 7p)
-        const isIOSWechat = WECHAT && this.os === OS.IOS;
-        const _tmpCanvas = document.createElement('canvas');  // TODO: remove this
-        let supportWebp: boolean;
-        if (isIOSWechat) {
-            supportWebp = true;
-        } else {
-            try {
-                supportWebp = TEST ? false : _tmpCanvas.toDataURL('image/webp').startsWith('data:image/webp');
-            } catch (e) {
-                supportWebp  = false;
-            }
+    private _supportsWebpPromise (): Promise<void> {
+        if (!TEST) {
+            return this._supportsWebp().then((isSupport) => {
+                this._setFeature(Feature.WEBP, isSupport);
+            });
         }
-        return supportWebp;
+        return Promise.resolve();
+    }
+
+    private _supportsWebp (): Promise<boolean> {
+        return new Promise<boolean>((resolve, reject) => {
+            const img = document.createElement('img');
+            const timer = setTimeout(() => {
+                resolve(false);
+            }, 500);
+            img.onload = function onload () {
+                clearTimeout(timer);
+                const result = (img.width > 0) && (img.height > 0);
+                resolve(result);
+            };
+            img.onerror = function onerror (err) {
+                clearTimeout(timer);
+                resolve(false);
+            };
+            img.src = 'data:image/webp;base64,UklGRiIAAABXRUJQVlA4IBYAAAAwAQCdASoBAAEADsD+JaQAA3AAAAAA';
```

Comment:
Could this work on minigame platforms ?
Does it change the behavior of `_supportsWebp ` method ?

Reply:
It works. Its working principle is to create a webp image, If it is created successfully, it means that the platform supports webp.